File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/masterdata/core/implementation/MasterDataServiceTest.java
package io.zenwave360.example.clinicaltool.modules.masterdata.core.implementation;

import io.zenwave360.example.clinicaltool.modules.masterdata.config.*;
import io.zenwave360.example.clinicaltool.modules.masterdata.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.masterdata.core.inbound.*;
import io.zenwave360.example.clinicaltool.modules.masterdata.core.inbound.dtos.*;
import io.zenwave360.example.clinicaltool.modules.masterdata.core.implementation.mappers.*;
import io.zenwave360.example.clinicaltool.modules.masterdata.core.outbound.jpa.*;
import io.zenwave360.example.clinicaltool.modules.masterdata.infrastructure.jpa.inmemory.*;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mapstruct.factory.Mappers;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.PageRequest;

import java.util.Map;
import java.util.Optional;
import java.time.*;
import java.math.BigDecimal;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.mockito.Mockito.*;

/**
 * Acceptance Test for MasterDataService.
 */
class MasterDataServiceTest  {

    private final Logger log = LoggerFactory.getLogger(getClass());

    ServicesInMemoryConfig context = new ServicesInMemoryConfig();
    MasterDataServiceImpl masterDataService = context.masterDataService();

    MasterDataRepositoryInMemory masterDataRepository = context.masterDataRepository();

	@BeforeEach
	void setUp() {
		context.reloadTestData();
	}



    @Test
    void createMasterDataTest() {
        MasterData input = null; // TODO
        // TODO fill input data
        // input.setType(MasterDataType.values()[0]);
        // input.setKey("");
        // input.setValue("");
        // input.setTranslations(List.of(new MasterDataTranslation()));
        var masterData = masterDataService.createMasterData(input);
        assertNotNull(masterData.getId());
        assertTrue(masterDataRepository.containsEntity(masterData));
}

    @Test
    void getMasterDataTest() {
        Long id = null;
        var masterData = masterDataService.getMasterData(id);
        assertTrue(masterData.isPresent());
}

    @Test
    void updateMasterDataTest() {
        Long id = null;
        MasterData input = null; // TODO
        // TODO fill input data
        // input.setType(MasterDataType.values()[0]);
        // input.setKey("");
        // input.setValue("");
        // input.setTranslations(List.of(new MasterDataTranslation()));
        // assertTrue(masterDataRepository.containsKey(id));
        var masterData = masterDataService.updateMasterData(id, input);
        assertTrue(masterData.isPresent());
        assertTrue(masterDataRepository.containsEntity(masterData.get()));
}

    @Test
    void listMasterDataTest() {
        // var results = masterDataService.listMasterData(PageRequest.of(0, 10));
        // assertNotNull(results);
}

    @Test
    void deleteMasterDataTest() {
        Long id = null;
        // assertTrue(masterDataRepository.containsKey(id));
        masterDataService.deleteMasterData(id);
        // assertFalse(masterDataRepository.containsKey(id));
}

    @Test
    void listMasterDataOfTypeTest() {// TODO: implement this test
}

}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/masterdata/MasterDataServiceTest.kt
package io.zenwave360.example.clinicaltool.modules.masterdata

import io.zenwave360.example.clinicaltool.modules.masterdata.*
import io.zenwave360.example.clinicaltool.modules.masterdata.config.*
import io.zenwave360.example.clinicaltool.modules.masterdata.domain.*
import io.zenwave360.example.clinicaltool.modules.masterdata.dtos.*
import io.zenwave360.example.clinicaltool.modules.masterdata.inmemory.*
import io.zenwave360.example.clinicaltool.modules.masterdata.mappers.*
import java.time.*
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.Mockito.*
import org.slf4j.Logger
import org.slf4j.LoggerFactory

/** Acceptance Test for MasterDataService. */
class MasterDataServiceTest {

    private val log: Logger = LoggerFactory.getLogger(javaClass)

    val context = ServicesInMemoryConfig()
    val masterDataService: MasterDataService = context.masterDataService()

    val masterDataRepository: MasterDataRepositoryInMemory = context.masterDataRepository()

    @BeforeEach
    fun setUp() {
        context.reloadTestData()
    }

    @Test
    fun createMasterDataTest() {
        val input: MasterData = MasterData() // TODO
        // TODO fill input data
        // input.type = MasterDataType.values()[0]
        // input.key = ""
        // input.value = ""
        // input.translations = List.of(MasterDataTranslation())
        val masterData = masterDataService.createMasterData(input)
        assertNotNull(masterData.id)
        assertTrue(masterDataRepository.containsEntity(masterData))
    }

    @Test
    fun getMasterDataTest() {
        val id: Long = 1L
        val masterData = masterDataService.getMasterData(id)
        assertNotNull(masterData)
    }

    @Test
    fun updateMasterDataTest() {
        val id: Long = 1L
        val input: MasterData = MasterData() // TODO
        // TODO fill input data
        // input.type = MasterDataType.values()[0]
        // input.key = ""
        // input.value = ""
        // input.translations = List.of(MasterDataTranslation())
        // assertTrue(masterDataRepository.containsKey(id))
        val masterData = masterDataService.updateMasterData(id, input)
        assertNotNull(masterData)
        assertTrue(masterDataRepository.containsEntity(masterData!!))
    }

    @Test
    fun listMasterDataTest() {
        // val results = masterDataService.listMasterData(PageRequest.of(0, 10))
        // assertNotNull(results)
    }

    @Test
    fun deleteMasterDataTest() {
        val id: Long = 1L
        // assertTrue(masterDataRepository.containsKey(id))
        masterDataService.deleteMasterData(id)
        // assertFalse(masterDataRepository.containsKey(id))
    }

    @Test
    fun listMasterDataOfTypeTest() { // TODO: implement this test
    }
}
